apps/prairielearn/src/lib/workspace.ts

```diff
@@ -431,7 +487,7 @@ async function initialize(workspace_id: string): Promise<InitializeResult> {
   ).filter((file): file is WorkspaceFile => !!file);
 
   const dynamicFiles: WorkspaceFile[] =
-    (variant.params?._workspace_files as DynamicWorkspaceFile[] | null)
+    (params?._workspace_files as DynamicWorkspaceFile[] | null)
```

Comment:
This is almost definitely overkill for this PR, but we should probably be treating `params._workspace_files` as `unknown` and proving to both ourselves and the compiler as we go that it's an array, that each entry is an object, that the object has a `name` property, that `name` is a string, and so on. Maybe we could leverage Zod here, but it might be more effort to extract useful entries for `fileGenerationErrors` from that...

Again, not requesting a change right now, just something that occurred as I was looking at this.

Reply:
Added as an issue in #9862.